// Copyright 2023 The ChromiumOS Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

// List of events triggered during the Rollback Process.
// TODO(b/261850979): Add Enterprise Rollback events.
enum EnterpriseRollbackEvent {
  EVENT_UNSPECIFIED = 0;
  ROLLBACK_POLICY_ACTIVATED = 1;
}

// Information about the triggered Rollback event required to log the metric.
message EventData {
  EnterpriseRollbackEvent event = 1;
  // TODO(b/261850979): add optional extra event details required for certain
  // metrics.
}

// EnterpriseRollbackMetricsData keeps track of the events related to the most
// recent Enterprise Rollback process. The header contains the information
// relevant to the duration of the whole process.
message EnterpriseRollbackMetricsData {
  // Header. This field is only set once per Enterprise Rollback.
  RollbackMetadata rollback_metadata = 1;
  // Events triggered during the rollback process.
  repeated EventData event_data = 2;
}

// Information that applies to the whole duration of Enterprise Rollback.
message RollbackMetadata {
  // ChromeOS version is a string with format "15270.0.0". As strings are not
  // fully supported in structured metrics, we encode each version as three
  // integers: [major, minor, patch].
  uint32 origin_chromeos_version_major = 1;
  uint32 origin_chromeos_version_minor = 2;
  uint32 origin_chromeos_version_patch = 3;
  uint32 target_chromeos_version_major = 4;
  uint32 target_chromeos_version_minor = 5;
  uint32 target_chromeos_version_patch = 6;
}
